flake.nix

```diff
@@ -0,0 +1,81 @@
+{
+  description = "Flake for building p4-fusion with openssl1.1 and openssl3, dynamically and statically";
```

Comment:
Curiosity, why do we need both ?

Reply:
Im curious now which comment covers this actually 😅 (and also which part you meant, openssl versions or dynamic+static?)

Reply:
Why would we need both versions of openssl AND dynamic + static. But below you explain that when targeting darwin-amd64 it's necessary, so I kinda stopped at that :)

Reply:
Oh I include openssl3 because technically 1.1 is EOL (and we shouldn't be using it, requires explicit opt-in with nix) so Ive included 3.0 so it's there and available. Im hoping it maybe encourages someone (Erik 👁️) to have us use 3.0 instead.

Realistically we should probably only have the static targets, the dynamic ones dont make sense outside of the machine they were built on. Mostly a relic of me first having worked on the dynamic targets (faster to build and test the CI workflows) and then adding static ones today

Reply:
The note about darwin-amd64 is an implementation detail around workarounds in the nix code specifically, but both openssl versions and both dynamic and static library linking work on it.